client/src/App.tsx

```diff
@@ -83,6 +88,136 @@ const App = () => {
         document.title = (startingIn > 0)
             ? `${name} starting in ${startingIn} minute${startingIn !== 1 ? 's' : ''}.`
             : `${name} ending in ${endingIn} minute${endingIn !== 1 ? 's' : ''}, started ${-startingIn} minute${startingIn !== -1 ? 's' : ''} ago.`
+
+        /** ====== begin favicon ====== */
+
+        const favicon = document.createElement('link');
+        favicon.setAttribute('rel', 'icon')
+        document.head.appendChild(favicon);
+        let numToShow = startingIn > 0 ? startingIn : endingIn;
+        const isSeconds = (numToShow == 1);
+        let seconds;
+        if (isSeconds){
+            seconds = 60 - ( date.diff(midnight, 'seconds') % 60);
+            numToShow = seconds;
+        }
+        // document.title = ['isSeconds: ', isSeconds, ' & numToShow: ', numToShow].join()
+        const color = parsePeriodColor(next[0], userData?.data() as UserData)
+        const FAVICON_SIZE = 32
+        const borderRadius = FAVICON_SIZE * 0.15
+        const sRadius = FAVICON_SIZE * 0.45 // radius for last seconds
+        //  create the canvas
+        const faviconCanvas = document.createElement('canvas')
+        faviconCanvas.width = FAVICON_SIZE
+        faviconCanvas.height = FAVICON_SIZE
+        const fc = faviconCanvas.getContext('2d')!
+
+        // configure it to look nice
+        fc.textAlign = 'center'
+        fc.textBaseline = 'middle'
+        fc.lineWidth = FAVICON_SIZE * 0.1
+        fc.lineJoin = 'round'
+        fc.lineCap = 'round'
+
+        // colourtoy
+
+        const colourtoy = document.createElement('div')
+        function isLight (colour : string) {
+            colourtoy.style.backgroundColor = colour
+            colour = colourtoy.style.backgroundColor
+            let colorArr:number[] = colour
+                .slice(colour.indexOf('(') + 1, colour.indexOf(')'))
+                .split(/,\s*/)
+                .map(a => +a)
+            // https://stackoverflow.com/questions/11867545/change-text-color-based-on-brightness-of-the-covered-background-area
+            return (
+                Math.round(
+                    (parseInt(''+colorArr[0]) * 299 +
+                        parseInt(''+colorArr[1]) * 587 +
+                        parseInt(''+colorArr[2]) * 114) /
+                    1000
+                ) > 150
+            )
+        }
```

Comment:
tfw UGWA code. Since y'all have NPM at your disposal, there's probably a better library for this since IIRC this formula is outdated